baselines/pFedHN/pFedHN/strategy.py

```diff
@@ -0,0 +1,211 @@
+    Scalar,
+    ndarrays_to_parameters,
+    parameters_to_ndarrays,
+)
+from flwr.server.client_manager import ClientManager
+from flwr.server.client_proxy import ClientProxy
+
+from pFedHN.models import CNNHyper
+
+
+# pylint: disable=invalid-name
+# pylint: disable=too-many-instance-attributes
+class pFedHN(fl.server.strategy.Strategy):
+    """Federated strategy with pFedHN."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        config,
+        fraction_fit: float = 0.0,
+        fraction_evaluate: float = 1.0,
+        min_fit_clients: int = 1,
+        min_evaluate_clients: int = 1,
+        min_available_clients: int = 1,
+        accept_failures: bool = True,
+        initial_parameters: Optional[Parameters] = None,
+        evaluate_fn: Optional[
+            Callable[
+                [int, NDArrays, Dict[str, Scalar]],
+                Optional[Tuple[float, Dict[str, Scalar]]],
+            ]
+        ] = None,
+    ) -> None:
+        super().__init__()
+        self.fraction_fit = fraction_fit
+        self.fraction_evaluate = fraction_evaluate
+        self.min_fit_clients = min_fit_clients
+        self.min_evaluate_clients = min_evaluate_clients
+        self.min_available_clients = min_available_clients
+        self.accept_failures = accept_failures
+        self.initial_parameters = initial_parameters
+        self.cfg = config
+        self.evaluate_fn = evaluate_fn
+        # Initialising our hnet model in server
+        self.hnet = CNNHyper(
+            n_nodes=self.cfg.client.num_nodes,
+            embedding_dim=int(1 + self.cfg.client.num_nodes / 4),
+            in_channels=self.cfg.model.in_channels,
+            n_kernels=self.cfg.model.n_kernels,
+            out_dim=self.cfg.model.out_dim,
+            hidden_dim=100,
+            n_hidden=1,
+        )
+
+    def __repr__(self) -> str:
+        """Return the strategy name."""
+        return "pFedHN"
+
+    def initialize_parameters(
+        self, client_manager: ClientManager
+    ) -> Optional[Parameters]:
+        """Initialize global model parameters."""
+        client_manager.num_available()
+
+        self.hnet.to(torch.device("cpu"))
+
+        initial_parameters = self.initial_parameters
+        self.initial_parameters = None
+        return initial_parameters
+
+    def num_fit_clients(self, num_available_clients: int) -> Tuple[int, int]:
+        """Return sample size and required number of clients."""
+        return min(1, num_available_clients), self.min_available_clients
+
+    def num_evaluate_clients(self, num_available_clients: int) -> Tuple[int, int]:
+        """Use a fraction of available clients for evaluation."""
+        return min(0, num_available_clients), self.min_available_clients
+
+    def configure_fit(
+        self, server_round: int, parameters: Parameters, client_manager: ClientManager
+    ) -> List[Tuple[ClientProxy, FitIns]]:
+        """Configure the next round of training."""
```

Comment:
Should we clearly state that, unlike most methods out there, `pFedHN` uses just one client per round? I suppose the code below works even if more than one client is sampled per round, does it? what about the aggregation part?

Reply:
Explained explicitly that we are taking only one client per round
as https://github.com/achiverram28/flower/blob/main/baselines/pFedHN/pFedHN/strategy.py#L65C9-L65C23 
takes min of 1 and available